src/brod_topic_subscriber.erl

```diff
@@ -357,7 +357,10 @@ handle_info({'DOWN', _Mref, process, Pid, Reason},
       %% not a consumer pid
       {noreply, State}
   end;
-handle_info(_Info, State) ->
+handle_info(Info, #state{cb_module = CbModule, cb_state = CbState} = State) ->
+  %% Any unhandled messages are forwarded to the callback module to
+  %% support arbitrary message-passing.
+  brod_utils:optional_callback(CbModule, handle_info, [Info, CbState], ok),
   {noreply, State}.
```

Comment:
Could this be just:

```erlang
brod_utils:optional_callback(CbModule, handle_info, [Info, CbState], {noreply, State})
```

I might be totally misunderstanding the code though, I'm missing a lot of background context. Same comment goes for other invocations of `optional_callback`.

Reply:
You are correct.
I followed the example on https://github.com/salemove/brod/blob/7ed6aa716b261571c7c8d3ffdb6ac64cecc29486/src/brod_topic_subscriber.erl#L386, but now that I double checked, the style various from module to module, so I'll switch to the shorthand.

Reply:
Ah, no-no, I need to do better to conform to expected handle_info semantics, I'll push a change soon.

Reply:
https://github.com/salemove/brod/compare/7ed6aa716b261571c7c8d3ffdb6ac64cecc29486..0db28028010ce99ebd29287329187a0e4c31c205

Fixed and changed the test so that handle_info callback emits a trace. 
The traces are checked at the end of the tests to assert "state continuity", where every trace must increase the `state` value. Failing to incorporate `NewCbState` now makes the test fail.

Reply:
Looks good 👍